### ISSUE TITLE
[2.7.3] Promise causes hangs with Playwright library

### ISSUE THREAD

**Issue by hexsprite** [meteor/meteor#12158]
OS: MacOS 12.5.1 (plus others)

[Playwright](https://github.com/microsoft/playwright) is a browser automation and testing library maintained by Microsoft, and created by former puppeteer engineers. Very popular project, 41.5K stars on Github.

# Expected Behavior

Simple behavior like navigating to a page should work properly when invoked from Meteor server-side code.

# Actual Behavior

Playwright instantiates the browser, the page navigation occurs, but the Promise never returns so the application hangs at that point.

# Technical notes

It seems like possibly the Promise implementation of Playwright is conflicting with Meteor's.

I created a reproduction that can be run in BOTH Meteor, or using plain Node.

When running in plain node it works perfectly, unless I import Fibers and use the Meteor Promise implementation in which case it demonstrates the "Actual Behavior" above. This is why I believe the source is of the issue is Meteor's promise implementation.

Playwright has a ManualPromise class which extends Promise. I don't know if this is source of the issue or not but I mentioned it because it seems like a possible source of problems.

https://github.com/microsoft/playwright/blob/main/packages/playwright-core/src/utils/manualPromise.ts

# Reproduction

https://github.com/hexsprite/meteor-issue-12158-playwright-promises

I've provided a reproduction showing how the promise behavior is different in plain Node (WORKS), Node + Meteor-promise (WORKS), Node + meteor-promise + Fibers (BROKEN), and in Meteor (BROKEN).

See also:
https://github.com/Meteor-Community-Packages/meteor-browser-tests/issues/43

**Comment by StorytellerCZ** [comment]
@hexsprite meteor-browser-tests recently had new releases which focused on fixing issues with drivers, can you please check if this is still relevant?

**Comment by hexsprite** [comment]
@StorytellerCZ my reproduction doesn't depend on meteor-browser-tests, working on that package just highlighted this issue.

I just tested my repro with Meteor 3.0 alpha and it passes fine, so I'm closing this.

The problem with Meteor 2.x is that Fibers patches Promise in a way that is incompatible with playwright. A good workaround for Meteor 2.x is to use a subprocess to playwright and return the results back to Meteor, which is what we did to support Playwright as a driver in meteor-browser-tests....
